### PR TITLE
Fix blind keyed accounts indexing in Config program

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Config program blindly indexes into keyed_accounts. A client can crash the leader by sending it an invalid transaction.

#### Summary of Changes

Same fix as in Vest, Stake and System - iterate instead of index.

cc: #6292

### CODE REVIEW

**Comment by garious** [comment]
cc @mvines

**Comment by garious** [comment]
The BPF loader is quite big. I'd like to leave it out of SLP1.  As it turns out, it has this same problem!

**Comment by mvines** [comment]
Yeah, BPF loader can certainly be excluded from SLP1.  Config can as well

**Comment by garious** [comment]
Currently, Vest depends on Config. I can drop that though.

**Comment by mvines** [comment]
Ah, got it.  That's why you care :)

**Comment by CriesofCarrots** [comment]
> Ah, got it. That's why you care :)

@mvines solana-cli `validator-info` also uses Config. Maybe not a necessity for SLP1, but it would be a nice-to-have for participating validators (vis-a-vis network explorer display)

**Comment by mvines** [comment]
little tricky to do that without the system program, we could bake that into the SLP1 genesis block though

File: programs/config_api/src/config_processor.rs [hunk 1]
```diff
@@ -5,6 +5,7 @@ use bincode::deserialize;
 use log::*;
 use solana_sdk::account::KeyedAccount;
 use solana_sdk::instruction::InstructionError;
+use solana_sdk::instruction_processor_utils::next_keyed_account;
```
Comment (jackcmay) [inline programs/config_api/src/config_processor.rs]: @garious Why isn't this in `solana_sdk::account`?

Comment (garious) [reply programs/config_api/src/config_processor.rs]: Because of its dependency on InstructionError

Comment (jackcmay) [reply programs/config_api/src/config_processor.rs]: @garious How about we split `InstructionError` our of `Instruction.rs` into`instruction_error.rs` and have `account.rs` depend on `instruction_error.rs`?

Comment (garious) [reply programs/config_api/src/config_processor.rs]: Why bother?  I'm not a fan of error modules. See result.rs in `solana_core` for an example of how badly that can go.

Comment (garious) [reply programs/config_api/src/config_processor.rs]: Currently the function is only used by instruction processors, so being in instruction_processor_utils seems like a fine place to land.

Comment (jackcmay) [reply programs/config_api/src/config_processor.rs]: Ok, sounds good

**Pull request merged**